sylt-compiler/src/preamble.lua

```diff
@@ -1,7 +1,22 @@
 -- Begin Sylt preamble
-__TEST = false
-
 -- Built in types
+
+-- THE nil-table
+__NIL = setmetatable( { "nil" }, { __tostring = function() return "nil" end } )
+
+__INDEX = function(o, i)
+    local m = getmetatable(o)
+    if m._type == "tuple" or m._type == "list" then
+        local e = o[i + 1]
+        assert(e ~= nil, "Tuple/list index out of range \"" .. i .. "\"")
```

Comment:
Is this a problem if we have optional types in a list?

Reply:
No - because this is after the type-system, we have another nil-value than Nil